### PR TITLE
feat: support do not create project directory functionality

### PR DESCRIPTION
**Pull request by damen6**
* feat: support do not create project directory functionality remove `location.path: .` inside pitfile.yml file and put the code logic to support DO_NOT_CREATE_PROJECT_DIR parameter in code level during deploy component process.
* Added a code logic to check if project directory does not exist, how to read the directory path properly 
* updated unit tests for new added code logic

### CODE REVIEW

File: k8s-deployer/scripts/start-example.sh [hunk 1]
```diff
@@ -60,6 +60,7 @@ fi
 if [ "${DO_NOT_CREATE_PROJECT_DIR}" == "" ]; then DO_NOT_CREATE_PROJECT_DIR="false"; fi
 if [ "${DO_NOT_CREATE_PROJECT_DIR}" == "true" ];
 then
+  export DO_NOT_CREATE_PROJECT_DIR
```
Comment (fmarek-kindred) [inline k8s-deployer/scripts/start-example.sh:63]: We do not need to export it. The scope of this variable is only the current script.

Comment (damen6) [reply k8s-deployer/scripts/start-example.sh:63]: Will fix it soon ~

Comment (damen6) [reply k8s-deployer/scripts/start-example.sh]: Issue fixed ~

File: k8s-deployer/src/deployer.ts [hunk 2]
```diff
@@ -210,11 +239,14 @@ export const deployComponent = async (
   } else {
     logger.info("Reading commit sha of local project: '%s'", appDir)
 
+    if (process.env.DO_NOT_CREATE_PROJECT_DIR === "true") spec.location.path = "."; // only if creating project directory is not required
```
Comment (fmarek-kindred) [inline k8s-deployer/src/deployer.ts:242]: @damen6  this line should be removed.

Comment (damen6) [reply k8s-deployer/src/deployer.ts:242]: Will fix it now ~

Comment (damen6) [reply k8s-deployer/src/deployer.ts]: Fixed ~

**Review by fmarek-kindred — APPROVED** [review]

**Pull request merged**